src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs

```diff
@@ -314,6 +314,20 @@ private ISmartTokenFormatter CreateSmartTokenFormatter(OptionSet optionSet, IEnu
             return changes;
         }
 
+        private IEnumerable<IFormattingRule> GetTypingRules(Document document, SyntaxNode root, int position)
+        {
+            // make auto formatting on typing around missing brace case better
```

Comment:
this comment isn't helpful.  How does this make "typing around missing brace case better"?

Reply:
I moved existing comments as they are.

Reply:
Uh... i don't know :)    that's why i'm asking.  I can't comment it because i don't understand why it's doing what it's doing.  how, specifically, is it making typing around missing braces better?  

> I moved existing comments as they are

I can't find any occurrences of "make auto formatting" in the codebase now, and this is the only case where ths is written in this PR.  Where was this "moved" from?

Reply:
this is where comments is
https://github.com/dotnet/roslyn/pull/26708/files#diff-71009298e9ba0addad9cd737f4eaf9d8L194

Reply:
I'm confused.  What you linked to has 30+ lines of comments explaining what it's tryign to do.  The code here has a single, cryptic line that just says "make auto formatting on typing around missing brace case better".  it doesn't explain 

1. what the "missing brace case" is.
2. how it makes it better.
3. why the logic in it supports what it is doing. 

Code can often be clear on its own.  And in that case, it doesn't really need comments.  But i look at thsi code and i have no idea what it is doing or why it is correct.  Can we please appropriate comment code like this so that reviewers can understand it as well as making it clear for future maintainers?  

Thanks!

Reply:
I am not sure what to tell. code is under start token formatter which is for auto formatting. if you don't know what smart token formatter and what auto formatting is , I am not sure why you are reviewing this code which looks like you don't know much then.

Reply:
> I am not sure what to tell.

I have indicated the issue.  The comment ```// make auto formatting on typing around missing brace case better``` is not a good comment.  I have explained why.  

> code is under start token formatter which is for auto formatting

That does not answer the question or address the points i made about this comment.  

>  I am not sure why you are reviewing this code which looks like you don't know much then.

The onus is on teh person writing the change to make things as clear as possible for the team which needs to review and maintain the code.  An initial PR can use reasonable judgement on the acceptable level of commenting.  However, if that level of commenting is insufficient for people reviewing the code, then things hsould be improved.

Note: i believe i'm the person on the IDE team who has done hte second-most work in this area *after you*.  I'm telling you that your level of commenting feels insufficient and unclear.  I have asked you pleasantly to expand on things and bring things up to help both the reviewer and to help with the maintainability of the code.  I believe these are reasonable requests and commensurate with what is commonly asked for from all contributers to the project.

Reply:
well, now we have system to resolve conflict. so tagging @jinujoseph 

> what the "missing brace case" is.
you asked what missing brace case is, so I gave you comment where it is explained. not sure why that is not sufficient. since I am not changing actual rule, but move it to right place which I again explained below why it is right place in the PR. I dont think that needs to put in the code since once code is changed, there is no longer no code to refer to as bad.

> how it makes it better.
to explain how it makes it better, it require what it used to do which once code is removed has no meaning, so I explained that in the PR description. not sure why you want that in the comment in the code that referring to code that is no longer there.

> why the logic in it supports what it is doing.
I think if shows what it is doing, if you think it is lacking, please be more specific, or you can just give me what you want, I can just put it there.

Reply:
> you asked what missing brace case is, so I gave you comment where it is explained. not sure why that is not sufficient.

Because in no way would i know about that when reading the comment: // make auto formatting on typing around missing brace case better

Reply:
you can go to the rule and it is right there. what you are saying is comment should be there in the order you read the code. if I read code in different order, then comment should be duplicated there as well?

Reply:
> not sure why you want that in the comment in the code

Because that's what people will be reading in the future... Please put yourself in my shoes for a moment.  Say i need to debug through this code 6-12 months from now.  Something has gone wrong and i step in here and i see this comment and this check.  Now i'm scratching my head tring to figure out what's going on and if this might be hte source of the problem i'm looking at.

Now, i have a few things that could help.  The most ideal would be clear explanations in the code of what's going on.  That allows me to understand while keeping me in context.

in the absence of that i must now do an external investigation of this code to try to figure out what's going on.  For example, i may have to now track this code back in git to figure out hte inception point, which can be brutally difficult at times, especially when code is moved *across* files.  This may or may not lead to me useful information.

As such, i far prefer when we try to make the code *itself* as clear as possible to help future maintainers.  Note that this is not a speculative concern.  As someone who had to try to track down many issues in formatting over hte years, this sort of approach to the code would have been *majorly* appreciated.

I do not feel like it is unwarranted or excessive to ask people making changes to complex and subtle parts of the system to then put in a little extra effort toward making the changes clearer.  it really goes a long way and is very appreciated.  Thanks much!

Reply:
> I think if shows what it is doing, 

I do not.  That's what i said from the beginning and why i've asked for more clarification.  I have spent the time outlining precisely what i think is lacking and why i do not find the existing comment sufficient.

> or you can just give me what you want, I can just put it there.

As i've mentioned, i cannot do this.  I've explained what i do not understand, and i've asked for the clarification from you so i can understand.  I've even asked for other things that would help (for example a test that demonstrates the difference).  

I cannot write an explanatory comment myself when my feedback is: i need a comment to help make it clear waht's going on because i do not understand it :)  

Thanks!

Reply:
I agree if there was no comments. but it is there in different place then you want. if users hit it they will look around and find the comment. I do that every time. every one does that every time. whenever people uses any shared lib, we do that every time. we don't say, comment is not right here so I can't figure out.

Reply:
> every one does that every time. whenever people uses any shared lib, we do that every time. we don't say, comment is not right here so I can't figure out.

This is not true.  In many PRs reviewers ask for clarity in different parts of the codebase.  Consider the feedback i gave to Ivan during his linq work, or the feedback i'm giving Alrz in the 'invert if' PR.  

As a reviewer i'm reading the code and making a determination on every line of it about:

1. do i understand what it's doing *now*.
2. do i think people would understand it in the future.

In *both* cases the answer is "no".  When i encounter that, i ask the person owning the review to solidify the code in that area.   This happens all the time in PRs and is an essential part of the process.